pypeit/core/gui/gui_util.py

```diff
@@ -0,0 +1,257 @@
+                or (event.name == f'{event_type}_press_event' and event.inaxes is not self.ax):
+            self.drag_active = False
+            event.canvas.release_mouse(self.ax)
+            return
+
+        self._set_event(getattr(event, event_type), event.xdata, event.ydata)
+
+    def _button_update(self, event):
+        """Execute an event created by a button press."""
+        self._event_update(event, 'button')
+
+    def _key_update(self, event):
+        """Execute an event created by a key or mouse press."""
+        self._event_update(event, 'key')
+
+    def _set_event(self, action, x, y):
+        """
+        Execute an event.
+
+        Args:
+            action (:obj:`str`):
+                The keyword for the event action to perform
+            x (:obj:`float`):
+                X position in the pyplot window where the event took place
+            y (:obj:`float`):
+                Y position in the pyplot window where the event took place
+        """
+        self.action = action
+        self.pos = (x, y)
+        if not self.eventson:
+            return
+        if self.action not in self.observers:
+            print(f'No action: {action} ({self.name}, {x}, {y})')
+            return
+        if self.action in self.observers:
+            self.observers[self.action](self.pos)
+
+    def register(self, action, func, descr=None):
+        """
+        Register a function to associate with a specific button or key press.
+
+        Args:
+            action (:obj:`str`):
+                The keyword for the event action to perform
+            func (callable):
+                The function to call when the corresponding event is triggered
+            descr (:obj:`str`, optional):
+                The description of the action being taken.  Used to build a help
+                dialog.
+        """
+        self.observers[action] = func
+        self.observers_descr[action] = descr
+
+    def disconnect(self, action):
+        """
+        Remove the action from the register.
+
+        Args:
+            action (:obj:`str`):
+                The keyword for the event action to remove
+        """
+        try:
+            del self.observers[action]
+        except KeyError:
+            pass
+        try:
+            del self.observers_descr[action]
+        except KeyError:
+            pass
+
+    def build_help(self):
+        """
+        Register the help dialog.  Any action already assigned to the '?' key
+        will be removed!
+        """
+        if '?' in self.observers:
+            msgs.warn('Help key (?) is already registered and will be overwritten')
+            self.disconnect('?')
+        self.register('?', self.print_help, descr='Print this list of key bindings')
+
+    def print_help(self, pos):
+        """
+        Print the help dialog.
+
+        Args:
+            pos (list):
+                Location where the help event was triggered.  This is ignored.
```

Comment:
I'm sure there's a good reason why `pos` is an argument (and then is ignored), but I flag it anyway!

Reply:
Yep, same response as to why we can't just pass pos[0] to some of the `EdgeInspectorGUI` functions.